### PR TITLE
Mayhem token is optional for mayhem.forallsecure.com

### PR DESCRIPTION
**Pull request by swarkentin**
When running a Github action, a `GITHUB_TOKEN` will automatically be provisioned
in the action runner environment. This token may be used instead of the Mayhem 
API token for uploading run results. This will enable the action to run for repository forks,
so that GitHub secrets do not need to be exposed.

If the action is used against a different Mayhem for Code URL, then the Mayhem
token will still be required.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -88,35 +86,12 @@ async function run(): Promise<void> {
...
-      run=$(${cli} run . ${argsString});
-      if [ -n "${sarifOutput}" ]; then
-        ${cli} wait $run -n ${account} --sarif ${sarifOutput}/target.sarif;
-        run_number=$(echo $run | awk -F/ '{print $NF}')
-        curl -H 'X-Mayhem-Token: token ${mayhemToken}' ${mayhemUrl}/api/v2/namespace/${account}/project/${project}/target/$fuzz_target/run/$run_number > mayhem.json
-      fi
+    fuzz_target=$(grep target: Mayhemfile | awk '{print $2}')
+    run=$(${cli} --verbosity debug run . ${argsString} -n ${account} --project ${repo.toLowerCase()});
+    if [ -n "${sarifOutput}" ]; then
+      ${cli} wait $run -n ${account} --sarif ${sarifOutput}/target.sarif;
```
Comment (swarkentin) [inline src/main.ts:92]: > --sarif ${sarifOutput}/target.sarif;

This doesn't look right ..  isn't `${sarifOutput}` already the file?

File: README.md [hunk 2]
```diff
@@ -18,13 +18,7 @@ Want to try it? [Get started for free](https://forallsecure.com/mayhem-free) tod
...
 To use the Mayhem for Code GitHub Action, perform the following steps:
 
-1. Create a Mayhem account and copy and paste your account token to GitHub Secrets.
-
-    a. Navigate to [mayhem.forallsecure.com](https://mayhem.forallsecure.com/) to register an account.
-
-    b. Click your profile drop-down and go to *Settings* > *API Tokens* to access your account API token.
-
-    c. Copy and paste your Mayhem token to your [GitHub Secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-an-organization).
+1.  Navigate to [mayhem.forallsecure.com](https://mayhem.forallsecure.com/) to register an account.
```
Comment (Andrew5194) [inline README.md:21]:
```suggestion
1. Navigate to [mayhem.forallsecure.com](https://mayhem.forallsecure.com/) to register an account.
```

File: README.md [hunk 3]
```diff
@@ -89,43 +83,43 @@ jobs:
...
 ```
 
 The mCode Action accepts the following inputs:
 
 | Required | Input Name | Type | Description | Default
 | --- | --- | --- | --- | ---
-| ✔️ | `mayhem-token` | string | Mayhem for Code account token |
+|   | `mayhem-url` | string | Path to a custom Mayhem for Code instance. | https://mayhem.forallsecure.com |
+|   | `mayhem-token` | string | Mayhem for Code account token. Only required if overriding `mayhem-url` |
 |   | `args` | string | Additional arguments such as specifying the `--corpus` directory path |
```
Comment (Andrew5194) [inline README.md:101]:
```suggestion
|   | `args` | string | Additional [arguments](https://mayhem.forallsecure.com/docs/mayhem-cli/getting-started/mayhem-cli-commands/#run) such as specifying the `--tests` directory path |
```

File: src/main.ts (hunk 1 above)
Comment (swarkentin) [reply src/main.ts:92]: undoing this change -- sarif needs to be a folder, not a file

File: README.md [hunk 4]
```diff
@@ -103,29 +96,30 @@ The mCode Action accepts the following inputs:
...
 
 ## Reports and GitHub Code Scanning
 
 Mayhem for Code generates [SARIF reports](https://sarifweb.azurewebsites.net/#:~:text=The%20Static%20Analysis%20Results%20Interchange,approved%20as%20an%20OASIS%20standard.) for your application security testing results.
 
-SARIF reports are generated using the `sarif-output` parameter, which specifies an output file path. Similarly, to upload the SARIF report to GitHub, use the `sarif_file` parameter to specify the location of an input SARIF file to upload to GitHub.
+SARIF reports are generated using the `sarif-output` parameter, which specifies an output file path.
 
-```sh
+To upload the SARIF report to GitHub, use the `github/codeql-action/upload-sarif@v1` action with the `sarif_file` parameter to specify the location of an input SARIF file to upload to GitHub.
```
Comment (swarkentin) [inline README.md:110]:
```suggestion
To upload the SARIF report to GitHub, use the `github/codeql-action/upload-sarif@v1` action with the `sarif_file` parameter to specify the location of a path containing SARIF results to upload to GitHub.
```

**Pull request merged**